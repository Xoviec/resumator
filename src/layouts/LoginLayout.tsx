import styled from "@emotion/styled";
import CssBaseline from "@mui/material/CssBaseline";
import { FunctionComponent } from "react";
import "../assets/css/global.css";
import LogoBlack from "../assets/images/iO-logo-black.png";

import { colors } from "../config/theme";

export const LoginLayout: FunctionComponent = ({ children }) => (
  <>
    <CssBaseline />
    <Layout bgColor={colors.lightGrey}>
      <Card>
        <LogoWrapper>
          <img src={LogoBlack} alt="logo" />
        </LogoWrapper>
        <div>{children}</div>
      </Card>
    </Layout>
  </>
);

interface LayoutProps {
  bgColor: string;
}

const Layout = styled.div<LayoutProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 15px;
  height: 100vh;
  background-color: ${({ bgColor }) => bgColor};
`;

const Card = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1.25rem;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 0.25rem;
  background-color: white;
`;

const LogoWrapper = styled.div`
  max-width: 200px;
`;
