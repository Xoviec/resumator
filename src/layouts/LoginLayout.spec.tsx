import { render } from "@testing-library/react";
import { LoginLayout } from "./LoginLayout";

describe("LoginLayout", () => {
  it("renders correctly", () => {
    const { container } = render(<LoginLayout />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders correctly with children", () => {
    const { container } = render(
      <LoginLayout>
        <div>test-content</div>
      </LoginLayout>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
