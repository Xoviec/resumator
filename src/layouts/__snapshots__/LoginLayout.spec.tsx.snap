// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginLayout renders correctly 1`] = `
<div
  class="css-va4j7j"
>
  <div
    class="css-nf597f"
  >
    <div
      class="css-ee05cj"
    >
      <img
        alt="logo"
        src="frontmenLogoIcon.png"
      />
    </div>
    <div />
  </div>
</div>
`;

exports[`LoginLayout renders correctly with children 1`] = `
<div
  class="css-va4j7j"
>
  <div
    class="css-nf597f"
  >
    <div
      class="css-ee05cj"
    >
      <img
        alt="logo"
        src="frontmenLogoIcon.png"
      />
    </div>
    <div>
      <div>
        test-content
      </div>
    </div>
  </div>
</div>
`;
